b2sdk/raw_simulator.py

```diff
@@ -48,6 +50,18 @@
 ]
 
 
+def get_bytes_range(data_bytes, bytes_range):
```

Comment:
inverting ifs in this function would let you avoid nesting and would make the code more readable
```python
if bytes_range is None:
    return data_bytes
if bytes_range[0] >  bytes_range[1]:
    raise UnsatisfiableRange()
if bytes_range[0] < 0:
    raise UnsatisfiableRange()
if bytes_range[1] > len(data_bytes):
    raise UnsatisfiableRange()
return data_bytes[bytes_range[0]:bytes_range[1]]
```
and you could also provide some detail in the exception so that the user gets something more useful than "something was wrong with a range"

Reply:
ehh... I just moved it to helper function and fixed logic with `bytes_range[1] > len(data_bytes)` - why it was so nested, and buggy after previous review...? :D :P

Reply:
because code gets resistant to review even faster than it gets resistant to testing. It's a tough situation.

Reply:
about
> and you could also provide some detail in the exception so that the user gets something more useful than "something was wrong with a range"
```
class UnsatisfiableRange(B2Error):
    def __str__(self):
        return "The range in the request is outside the size of the file"
```
so I reject :)

and really - this is for tests purposes - so one would see traceback and should be able to understand what happened...

Reply:
the user may pass a range to download (though this feature is not exposed by the CLI)